service/src/test/java/bio/terra/pipelines/common/utils/pagination/CursorBasedPageableTest.java

```diff
@@ -0,0 +1,115 @@
+    assertThrows(IllegalArgumentException.class, () -> cursorBasedPageable.getDecodedCursor(""));
+    assertThrows(
+        IllegalArgumentException.class, () -> cursorBasedPageable.getDecodedCursor("    "));
+  }
+
+  @Test
+  void encodeValue() {
+    CursorBasedPageable cursorBasedPageable = new CursorBasedPageable(10, null, null);
+
+    // because this uses a LocalDateTime.now() as part of its encoding strategy, its hard to test
+    // the full string matches so we are just testing the first 10 characters as a "good enough"
+    // test
+    String structuredValue = "###" + 10 + "### - " + LocalDateTime.now();
+    String encodedString = Base64.getEncoder().encodeToString(structuredValue.getBytes());
+    assertEquals(
+        encodedString.substring(0, 10),
+        cursorBasedPageable.getEncodedCursor("10", true).substring(0, 10));
+  }
+
+  @Test
+  void encodeValueNoPreviousOrNextPages() {
+    CursorBasedPageable cursorBasedPageable = new CursorBasedPageable(10, null, null);
+    assertFalse(cursorBasedPageable.hasCursors());
+    assertNull(cursorBasedPageable.getEncodedCursor("10", false));
+  }
+
+  @Test
+  void hasNextCursor() {
+    CursorBasedPageable cursorBasedPageableNoNext = new CursorBasedPageable(10, null, "blah");
+    Assertions.assertFalse(cursorBasedPageableNoNext.hasNextPageCursor());
+    assertTrue(cursorBasedPageableNoNext.hasCursors());
+    CursorBasedPageable cursorBasedPageableNext = new CursorBasedPageable(10, "balhblah", null);
+    assertTrue(cursorBasedPageableNext.hasNextPageCursor());
+    assertEquals("balhblah", cursorBasedPageableNext.getNextPageCursor());
+  }
+
+  @Test
+  void hasPrevCursor() {
+    CursorBasedPageable cursorBasedPageableNoPrev = new CursorBasedPageable(10, "blah", null);
+    Assertions.assertFalse(cursorBasedPageableNoPrev.hasPrevPageCursor());
+    assertTrue(cursorBasedPageableNoPrev.hasCursors());
+    CursorBasedPageable cursorBasedPageablePrev = new CursorBasedPageable(10, null, "balhblah");
+    assertTrue(cursorBasedPageablePrev.hasPrevPageCursor());
+    assertEquals("balhblah", cursorBasedPageablePrev.getPrevPageCursor());
+  }
+
+  @Test
+  void getSearchValue() {
+    CursorBasedPageable cursorBasedPageableNoCursor = new CursorBasedPageable(10, null, null);
+    assertEquals(
+        CursorBasedPageable.POSTGRES_INT4_MAX_VALUE, cursorBasedPageableNoCursor.getSearchValue());
+    CursorBasedPageable cursorBasedPageable = new CursorBasedPageable(10, encodedTenValue, null);
+    assertEquals("10", cursorBasedPageable.getSearchValue());
+    cursorBasedPageable = new CursorBasedPageable(10, null, encodedTenValue);
+    assertEquals("10", cursorBasedPageable.getSearchValue());
+  }
+
+  @Test
+  void cursorBasedPageableHashCode() {
+    CursorBasedPageable cursorBasedPageableNextCursor =
+        new CursorBasedPageable(10, "hmmm_ok", null);
+    assertEquals(
+        new HashCodeBuilder(17, 31).append(10).append("hmmm_ok").append((String) null).toHashCode(),
```

Comment:
minor: could add a comment that 17 and 31 are hardcoded in the specification classes' hashCode methods (same applies to the other test classes that do this kind of test)